src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTestsUsingCertWithPrivateKey.cs

```diff
@@ -67,6 +67,31 @@ public static void DecryptMultipleRecipients()
             Assert.Equal<byte>(content, contentInfo.Content);
         }
 
+        [Fact]
+        public static void DecryptSuccesfullyWithWrongKeyProducesInvalidSymmetricKey()
+        {
+            using (X509Certificate2 wrongRecipient = Certificates.RSAKeyTransfer5_ExplicitSkiOfRSAKeyTransfer4.TryGetCertificateWithPrivateKey())
+            {
+                string encryptedContent =
+"3082018806092A864886F70D010703A082017930820175020102318201303082" +
```

Comment:
I produced this by running the originally failing test in a loop until it reproduced the behavior and dumped the enveloped data in the fail case to make a stable test case for this.